sdk/python/kfp/_client.py

```diff
@@ -1280,11 +1282,25 @@ def upload_pipeline_version(self,
 
         if pipeline_name:
             pipeline_id = self.get_pipeline_id(pipeline_name)
-
-        response = self._upload_api.upload_pipeline_version(
-            pipeline_package_path,
+        kwargs = dict(
             name=pipeline_version_name,
-            pipelineid=pipeline_id)
+            pipelineid=pipeline_id,
+        )
+
+        if description:
+            kwargs['description'] = description
+        try:
+            response = self._upload_api.upload_pipeline_version(
+                pipeline_package_path, **kwargs)
+        except kfp_server_api.exceptions.ApiTypeError as e:
+            # ToDo: Remove this once we drop support for kfp_server_api < 1.7
+            if 'description' in e.message:
+                logging.warning(
```

Comment:
I am not sure of what others think, but my understanding is to just throw this error message.
Because it's better the user upgrades KFP server api when they see this error.

Reply:
Yes, on a second thought, error might be better.

Reply:
Adding a NotImplementedError?